src/compiler/scala/tools/nsc/transform/SpecializeTypes.scala

```diff
@@ -1914,11 +1925,10 @@ abstract class SpecializeTypes extends InfoTransform with TypingTransformers {
 //        for (m <- specialOverrides(specializedClass)) specializedClass.info.decls.enter(m)
       val mbrs = new mutable.ListBuffer[Tree]
       var hasSpecializedFields = false
-
-      for (m <- sClass.info.decls
-             if m.hasFlag(SPECIALIZED)
-                 && (m.sourceFile ne null)
-                 && satisfiable(typeEnv(m), !sClass.hasFlag(SPECIALIZED))) {
+      val declsToSpecialize = sClass.info.decls.iterator.filter(m =>
```

Comment:
`declsToSpecialize` is not used anywhere but in the `for` comprehension, so perhaps this can also be fused.